packages/astro/src/core/add/index.ts

```diff
@@ -0,0 +1,417 @@
+import type yargs from 'yargs-parser';
+import path from 'path';
+import fs from 'fs/promises';
+import { execaCommand } from 'execa';
+import { fileURLToPath } from 'url';
+import { diffWords } from 'diff';
+import boxen from 'boxen';
+import prompts from 'prompts';
+import preferredPM from 'preferred-pm';
+import ora from 'ora';
+import { resolveConfigURL } from '../config.js';
+import { apply as applyPolyfill } from '../polyfill.js';
+import { error, info, debug, LogOptions } from '../logger.js';
+import * as msg from '../messages.js';
+import { dim, red, cyan, green, magenta, bold, reset } from 'kleur/colors';
+import { parseNpmName } from '../util.js';
+import { wrapDefaultExport } from './wrapper.js';
+import { ensureImport } from './imports.js';
+import { t, parse, visit, generate } from './babel.js';
+
+export interface AddOptions {
+	logging: LogOptions;
+	cwd?: string;
+	flags: yargs.Arguments;
+}
+
+export interface IntegrationInfo {
+	id: string;
+	packageName: string;
+	dependencies: [name: string, version: string][];
+}
+
+const DEFAULT_FRAMEWORKS = [
+	{ value: 'react', title: 'React' },
+	{ value: 'preact', title: 'Preact' },
+	{ value: 'vue', title: 'Vue' },
+	{ value: 'svelte', title: 'Svelte' },
+	{ value: 'solid-js', title: 'Solid' },
+	{ value: 'lit', title: 'Lit' },
+]
+const DEFAULT_ADDONS = [
+	{ value: 'tailwind', title: 'Tailwind' },
+	{ value: 'turbolinks', title: 'Turbolinks' },
+	{ value: 'partytown', title: 'Partytown' },
+	{ value: 'sitemap', title: 'Sitemap' },
+]
+
+const ALIASES = new Map([
+	['solid', 'solid-js'],
+	['tailwindcss', 'tailwind'],
+])
+const INSIGNIFICANT_CHARS = new Set([',', ']', '}'])
```

Comment:
We could move this to a JSON file somewhere, like `packages/integrations/manifest.json`. This file could be used by `create-astro` too

Reply:
A great idea! I'm not sure if a separate package is in scope for this PR, but it's a good follow-up step.